### ISSUE TITLE
Understanding STGCN implementation

### ISSUE THREAD

**Issue by nyanmn** [open-mmlab/mmaction2#1480]
I am looking at STGCN implementation and have two questions.
(1)Now number of person in a frame is 2. My application, I can have varying number of people in a frame. How can I do that?
(2)In the paper, there are three partitioning approaches were discussed.
Where they are implemented in program?
Is the following line implement one of the partitioning approaches
`x = torch.einsum('nkctv,kvw->nctw', (x, adj_mat))` line 163 of stgcn.py

**Comment by gengenkai** [comment]
> he paper, there are three partitioning approaches were discussed.
> Where they are implemented in program?

Hi, nyanmn.
1) In this case, you can modify the number of person(2) to the maximum number of people in your dataset. If a frame contain less than max_num people, you can use zero padding.
2) The partition strategy is implemented in models/skeleton_gcn/utils/graph.py.

**Comment by nyanmn** [comment]
Go through all source codes (base.py, skeletongcn.py, stgcn.py) for setting up and training parts.
But can't find where this 'number of instance in a frame' is defined.

skeletons = data_batch['keypoint'] in base.py line number 148.
This data_batch is the output of dataloader.
I looked through dataloader setup, can't find the initialization of 'number of instance in a frame'

**Comment by nyanmn** [comment]
Training part is still the same because those activities in NTURGB+D 60 or NTURGB+D 120 has one or two persons in actions.
Because most of the actions have maximum two persons only except group fighting.
But detection part is different.
In the detection video scene there may be multiple people like 7,8,9,10. But two persons are doing hand shaking and like to detect that hand shaking activity. How can I do that?  How can I select these two persons from the group of people?

**Comment by gengenkai** [comment]
> shaking activity. How can I do that? How can I select these two persons from the group of people?

Hi, the scene you mentioned is very common and I think it is kind of difficult for skeleton-based models to only recognize the 'fighting' action unless you have some ideas to only select these two people who are fighting and filter other unrelated people in this complicated scene. 
I suggest you can set the labels of this scenes as [walks(other people), fight] instead of [fight]. Or  try to add more clean training data where only fighting people exists in the video maybe help you solve this problem.
